src/lib/mongoBackend/mongoGetSubscriptions.cpp

```diff
@@ -123,10 +123,10 @@ static void setNotification(Subscription* s, const BSONObj& r)
   s->notification.throttling = r.hasField(CSUB_THROTTLING)? getField(r, CSUB_THROTTLING).numberLong() : -1;
 
   // Last Notification
-  s->notification.lastNotification = r.hasField(CSUB_LASTNOTIFICATION) ? getField(r, CSUB_LASTNOTIFICATION).numberLong() : -1;
+  s->notification.lastNotification = getLongField(r, CSUB_LASTNOTIFICATION);
```

Comment:
This should be:

```
s->notification.lastNotification = r.hasField(CSUB_LASTNOTIFICATION) ? getLongField(r, CSUB_LASTNOTIFICATION) : -1;
```

From a caller point of view, you have to be sure that the field exists when invoking get*Field() fucntion in safeBson module. Don't rely in the function returning a -1, as it also generate a Runtime Error.